### PR TITLE
ALAMO pyomo expression generation

### PR DESCRIPTION
**Pull request by mengleCMU**
## Summary/Motivation:
Generates pyomo expressions for use in pyomo models without the need for an import statement. A test example uses the pyomo expression in a toy pyomo model.

## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by osarwar — APPROVED** [review]

**Comment by MAZamarripa** [comment]
@mengleCMU we need to meet, I get some errors when trying an idaes model (indexed variables)

File: idaes/surrogate/alamopy/tests/test_mi.py [hunk 1]
```diff
@@ -61,6 +63,54 @@ def test_multiple_input():
 
     res = alamopy.doalamo(x, z)
 
+@pytest.mark.unit
+def test_gen_pyomo_model():
+
+    if not alamopy.has_alamo():
+        return False
```
Comment (andrewlee94) [inline idaes/surrogate/alamopy/tests/test_mi.py:70]: I think this would be better done using `pytest.mark.skipif`.

**Comment by ksbeattie** [comment]
@mengleCMU is working on this, @ksbeattie and @lbianchi-lbl will work on a test platform that have ALAMO & BARON installed.

**Comment by lbianchi-lbl** [comment]
@mengleCMU we're looking at open PRs in preparation for the August release. Is this work expected to be completed in the next few days? Otherwise, we can keep tracking this PR as part of the November release. @ksbeattie and I are making progress on the ALAMO & BARON support for the testing environment and we're planning to start introducing it after the August release anyways.

**Comment by ksbeattie** [comment]
This will likely be obsoleted by #455

**Comment by andrewlee94** [comment]
This is obsolete with progress going forwards on PR #455.

**Pull request closed without merging**